dspace-api/src/main/java/org/dspace/util/ProxyUtils.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.util;
+
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpHost;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+
+/**
+ * Utility methods for HTTP proxies.
+ * See configuration properties {@code http.proxy.host}, {@code http.proxy.port}.
+ * If host is set and port is not, or if host is not set, then proxying is
+ * disabled.
+ *
+ * @author mwood
+ */
+public class ProxyUtils {
+    public static final String C_HTTP_PROXY_HOST = "http.proxy.host";
+    public static final String C_HTTP_PROXY_PORT = "http.proxy.port";
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private ProxyUtils() {}
+
+    /**
+     * Add a proxy to an HttpClientBuilder if a proxy is configured.
+     *
+     * @param builder the builder to be configured.
+     * @return the builder, possibly configured with a proxy.
+     */
+    public static HttpClientBuilder addProxy(HttpClientBuilder builder) {
```

Comment:
Couldn't we consider refactoring this to be:
```
public static HttpHost getProxyHttpHost() {
   ...(current logic)..

   return new HttpHost(proxyHost, proxyPort);
}
```

Then we could just pass it into the builder directly like this:
```
HttpClient httpClient = HttpClientBuilder.create().addProxy(ProxyUtils.getProxyHttpHost()).build();
```

This might simplify some of other code you had to refactor, as all of our HttpClientBuilders tend to be built using chained code like that.

Is there a reason why you wanted this to modify the HttpClientBuilder directly, instead of using the `HttpClientBuilder.setProxy(HttpHost)` inline?